### ISSUE TITLE
Don't reconnect to all connections every time the lookupd job is run

### ISSUE THREAD

**Issue by werkshy** [nsqio/nsq-java#4]
It looks like we call reader.connectToNsqd() for every producer on every poll of lookupd, and we never disconnect from producers that lookupd doesn't respond to.

It seems like we should compare the list that lookupd returns to the list we have open, and open/close connections until we are reconciled.

**Comment by danielhfrank** [comment]
Yeah, that's correct. Clearly I wrote that only thinking about the initial connection. Will fix once I bring your pull req in

**Comment by danielhfrank** [comment]
Err, I see you may have actually done some work on this one already. We'll get to it next
